src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs

```diff
@@ -10,42 +15,191 @@ namespace Infrastructure.Common
         }
 
         public static bool Domain_Joined()
         {
-            return IsConditionalTestEnabled(TestProperties.Domain_Joined_PropertyName);
+            // If TestProperties does not specify true/false explicitly, we need to autodetect whether
+            // the client and server are domain joined.
+            // Temporarily use the simple heuristic that if we are running the services locally, it is.
+            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
+            return GetConditionValueFromTestProperties(TestProperties.Domain_Joined_PropertyName,
+                                                       Detect_Server_Is_Local);
         }
 
         public static bool Root_Certificate_Installed()
         {
-            return IsConditionalTestEnabled(TestProperties.Root_Certificate_Installed_PropertyName);
+            // Give precedence to TestProperties, but if the value is unspecified, go ahead and
+            // try to install the root cert now.  Failure auto-detects it cannot be done and skips
+            // subsequent tests.
+            return GetConditionValueFromTestProperties(TestProperties.Root_Certificate_Installed_PropertyName,
+                                                       () =>
+                                                       {
+                                                           try
+                                                           {
+                                                               BridgeClientCertificateManager.InstallRootCertificateFromBridge();
+                                                               return true;
+                                                           }
+                                                           catch
+                                                           {
+                                                               return false;
+                                                           }
+                                                       });
         }
 
         public static bool Client_Certificate_Installed()
         {
-            return IsConditionalTestEnabled(TestProperties.Client_Certificate_Installed_PropertyName);
+            // Give precedence to TestProperties, but if the value is unspecified, go ahead and
+            // try to install the local cert now.  Failure auto-detects it cannot be done and skips
+            // subsequent tests.
+            return GetConditionValueFromTestProperties(TestProperties.Client_Certificate_Installed_PropertyName,
+                                                       () =>
+                                                       {
+                                                           try
+                                                           {
+                                                               BridgeClientCertificateManager.InstallLocalCertificateFromBridge();
```

Comment:
As I mentioned I understand it is because we don't have a pre-run setup logic in current testing systems (TOF or Helix), I am okay your current proposal as a temporary solution for us.